test/IISIntegration.FunctionalTests/Utilities/IISApplication.cs

```diff
@@ -0,0 +1,249 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Server.IntegrationTesting.Common;
+using Microsoft.Extensions.Logging;
+using Microsoft.Web.Administration;
+
+namespace Microsoft.AspNetCore.Server.IntegrationTesting
+{
+    /// <summary>
+    /// Represents the IIS website registered in the global applicationHost.config
+    /// </summary>
+    internal class IISApplication
+    {
+        private const int IISRetryLimit = 500;
+        private const int IISWaitMillisecondsTimeout = 10;
+        private readonly ServerManager _serverManager = new ServerManager();
+        private readonly DeploymentParameters _deploymentParameters;
+        private readonly ILogger _logger;
+        private readonly string _ancmVersion;
+        private readonly object _syncLock = new object();
+        private readonly string _apphostConfigPath = Path.Combine(Environment.SystemDirectory, "inetsrv", "config", "applicationhost.config");
```

Comment:
Are we planning on doing x64/x86 testing?

Reply:
Yes, but the config is shared across 32 and 64 bit for IIS.